### ISSUE TITLE
[INFRA/IIGO] nans in broadcastTaxation

### ISSUE THREAD

**Issue by Eirikalb** [SOMAS2020/SOMAS2020#343]
Nans in BroadcastTaxation.

Reproduce by running: https://somas2020.github.io/SOMAS2020/#/newrun, new run and check logs if you get an oh REEEEEEE message:

![image](https://user-images.githubusercontent.com/25349833/104009215-f57f8f80-51aa-11eb-8ebd-bd2ce089e588.png)

# Log
````
2021/01/08 12:07:18 [SERVER]: start runTurn
2021/01/08 12:07:18 [SERVER]: TURN: 1, Season: 1
2021/01/08 12:07:18 [SERVER]: start startOfTurn
2021/01/08 12:07:18 [Team6]: Agent[Team6] configuration has been updated
2021/01/08 12:07:18 [Team6]: Agent[Team6] friendship information has been updated
2021/01/08 12:07:18 [Team6]: There are [6] islands left in this game
2021/01/08 12:07:18 [Team1]: Emotional state: Normal
2021/01/08 12:07:18 [Team1]: Resources: 100
2021/01/08 12:07:18 [Team3]: Our Status: {Resources:100 LifeStatus:Alive CriticalConsecutiveTurnsCounter:0}
2021/01/08 12:07:18 [SERVER]: finish startOfTurn
2021/01/08 12:07:18 [SERVER]: start runOrgs
2021/01/08 12:07:18 [SERVER]: start runIIGO
2021/01/08 12:07:18 [SERVER]: President Team3, Speaker Team1, Judge Team2
2021/01/08 12:07:18 [SERVER]: [EXECUTIVE]: Tax: map[Team1:NaN Team2:NaN Team3:NaN Team4:NaN Team5:NaN Team6:NaN]
2021/01/08 12:07:18 [SERVER]: [EXECUTIVE]: Resource Allocation: map[Team1:-0 Team2:-0 Team3:56.00000001259784 Team4:-6.2989156481913814e-09 Team5:-6.2989156481913814e-09 Team6:-0]
````

**Comment by Eirikalb** [comment]
@Fabio752  and @akowal3 seems to be authors. @Spruudis or @Neelesh99 could one of you fix this error?

**Comment by Neelesh99** [comment]
We're patching all global vars right now, hopefully, this will patch the NaN issue as @lhl2617 mentioned. If not I'll forward to @Fabio752 and @akowal3.

**Comment by Eirikalb** [comment]
Cheers, sweeet!

**Comment by lhl2617** [comment]
This is because globals are not reset

**Comment by lhl2617** [comment]
this issue should be resolved by #357 . @Eirikalb can you check if this can be repro

**Comment by Eirikalb** [comment]
> this issue should be resolved by #357 . @Eirikalb can you check if this can be repro

@lhl2617 not on my computer currently so unable to repro, won't be able to write anything the next 2/3 hours

**Comment by Eirikalb** [comment]
@lhl2617 tried reproducing cause and log showed tax in first round be real values, so AFAIK the issue is resolved
